casper/src/main/scala/coop/rchain/casper/merging/ConflictSetMerger.scala

```diff
@@ -44,11 +50,65 @@ object ConflictSetMerger {
         .getOrElse(Set.empty)
     }
 
+    def calMergedResult(
+        branch: Branch,
+        originResult: Map[Blake2b256Hash, Long]
+    ): Option[Map[Blake2b256Hash, Long]] = {
+      val diff = branch.map(mergeableChannels).toList.combineAll
+      diff.foldLeft[Option[Map[Blake2b256Hash, Long]]](Some(originResult)) {
+        case (baOpt, br) =>
+          baOpt.flatMap {
+            case ba =>
+              try {
+                val result = Math.addExact(ba.getOrElse(br._1, 0L), br._2)
+                if (result < 0) {
+                  none
+                } else {
+                  Some(ba.updated(br._1, result))
+                }
+              } catch {
+                case _: ArithmeticException => none
+              }
+          }
+
+      }
+    }
+
+    def foldRejection(baseBalance: Map[Blake2b256Hash, Long], branches: Set[Branch]) = {
+      val (_, rejected) = branches.foldLeft((baseBalance, Set.empty[Branch])) {
+        case ((balances, rejected), deploy) =>
+          // TODO come up with a better algorithm to solve below case
+          // currently we are accumulating result from some order and reject the deploy once negative result happens
+          // which doesn't seem perfect cases below
+          //
+          // base result 10 and folding the result from order like [-10, -1, 20]
+          // which on the second case `-1`, the calculation currently would reject it because the result turns
+          // into negative.However, if you look at the all the item view 10 - 10 -1 + 20 is not negative
```

Comment:
As a simple solution can we sort by diffs, e.g. `20, -1, -10` so that increasing of balance is calculated first?

Reply:
That sounds better!! Overflow should happen fewer than negative!

Reply:
After some digging, I found for a branch the `NumberChannelsDiff` would always be `A: -10, B:+10`because of the nature of transferring which the sorting doesn't actually help too much.

Reply:
Yes, there can be multiple transfers in a branch so logic of sorting is more complex. Let's investigate this as part of another PR.